internal/component/otelcol/consumer.go

```diff
@@ -12,6 +12,11 @@ type Consumer interface {
 	otelconsumer.Logs
 }
 
+type ConsumerWithComponentID interface {
+	Consumer
+	ComponentID() string
+}
```

Comment:
```suggestion
// ComponentMetadata can be implemented by, for example, consumers exported by components, to provide the ID of the component which is exporting given consumer. This is used for Live Graph / Live Debugging features.
type ComponentMetadata interface {
	ComponentID() string
}
```
I think that's pretty good. I also don't think it needs to be a `Consumer`.